utils/filesystem/files.go

```diff
@@ -1160,6 +1205,23 @@ func (fs *VFS) LsFromOpenedDirectory(dir File) ([]string, error) {
 	return dir.Readdirnames(-1)
 }
 
+func (fs *VFS) LsRecursiveFromOpenedDirectory(ctx context.Context, dir File) (files []string, err error) {
+	err = parallelisation.DetermineContextError(ctx)
+	if err != nil {
+		return
+	}
+	err = fs.checkWhetherUnderlyingResourceIsClosed()
+	if err != nil {
+		return nil, fmt.Errorf("%w: underlying resource is closed", commonerrors.ErrUndefined)
+	}
+	if dir == nil {
+		return nil, fmt.Errorf("%w: nil directory", commonerrors.ErrUndefined)
+	}
```

Comment:
Move this to just after the DetermineContextError part.

This is like a check that the arguments are valid so it should be before the business logic. I think the code checking the arguments should be kept together.

Also I think the error description should not mention nil, say something like fmt.Errorf("%w: supplied directory was undefined", commonerrors.ErrUndefined)